app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java

```diff
@@ -8,55 +9,82 @@
 import io.intrepid.contest.base.PresenterConfiguration;
 import io.intrepid.contest.models.Category;
 import io.intrepid.contest.models.Contest;
+import timber.log.Timber;
 
 class NewContestPresenter extends BasePresenter<NewContestMvpContract.View> implements NewContestMvpContract.Presenter {
     public Contest.Builder contest;
-    private int screenIndex = 0;
 
     NewContestPresenter(@NonNull NewContestMvpContract.View view,
-                               @NonNull PresenterConfiguration configuration) {
+                        @NonNull PresenterConfiguration configuration) {
         super(view, configuration);
+        contest = new Contest.Builder();
     }
 
     @Override
     public void onViewCreated() {
         super.onViewCreated();
         contest = new Contest.Builder();
-        view.showContestSubmissionPage(screenIndex);
+        view.initializePages(contest);
+        view.showContestSubmissionPage(0);
     }
 
     @Override
     public void onNextButtonClicked() {
-        ContestCreationFragment childFragment = view.getChildEditFragment(screenIndex);
-        if (childFragment != null) {
-            childFragment.onNextClicked();
-        }
+        int currentIndex = view.getCurrentIndex();
+        ContestCreationFragment fragment = view.getChildEditFragment(currentIndex);
+        fragment.onNextClicked();
     }
 
     @Override
     public void onBackButtonClicked() {
-        if (screenIndex == 0) {
+        int currentIndex = view.getCurrentIndex();
+        if (currentIndex == 0) {
             view.cancelEdit();
         } else {
-            view.showContestSubmissionPage(--screenIndex);
+            view.showContestSubmissionPage(currentIndex - 1);
         }
     }
 
     @Override
     public void setContestName(String contestName) {
         contest.setTitle(contestName);
-        view.showContestSubmissionPage(++screenIndex);
+        view.setNextVisible(false);
+        showNextScreen();
+    }
+
+    private void showNextScreen() {
+        int currentIndex = view.getCurrentIndex();
+        view.showContestSubmissionPage(currentIndex + 1);
     }
 
     @Override
     public void setContestDescription(String contestDescription) {
         contest.setDescription(contestDescription);
-        view.showContestSubmissionPage(++screenIndex);
+        showNextScreen();
     }
 
     @Override
     public void setCategories(List<Category> categories) {
         contest.categories = categories;
         view.completeEditForm(contest.build());
     }
+
+    @Override
+    public void onNextStatusChanged(boolean nextEnabled) {
```

Comment:
I'm still looking for a good name to call this. Before I had two method: onNextValidated() and onNextInvalidated(). The objective was to consolidate both into a single method that takes two booleans.What if I name this onNextValidated(boolean nextValidated)